apis/kueue/v1beta1/workload_types.go

```diff
@@ -603,6 +603,7 @@ const (
 // +kubebuilder:resource:shortName={wl}
 
 // Workload is the Schema for the workloads API
+// +kubebuilder:validation:XValidation:rule="self.metadata.labels['kueue.x-k8s.io/workload-uid'] == oldSelf.metadata.labels['kueue.x-k8s.io/workload-uid']", message="field is immutable"
```

Comment:
This probably needs to be relaxed to "field is immutable once set" since this is set in workload_reconciler. 
The bonus question: could it be set on workload creation with [Mutating Admission Policy](https://kubernetes.io/docs/reference/access-authn-authz/mutating-admission-policy/) ( I know it is still Alpha, but we could already plan to move the setting code).

Reply:
Yeah, I don't think you can test it at the UT level, because it is enforced by api-server. 

We cover other validation rules at the level of integration tests [example](https://github.com/kubernetes-sigs/kueue/blob/89f1497e552150eff12e9e30b4e28cf0dafb00fb/test/integration/singlecluster/tas/tas_test.go#L124-L136)

Reply:
Added integration tests